### PR TITLE
fileinstall: fix

### PR DESCRIPTION
**Pull request by oliver-sanders**
This test is failing on 2019.01.x:
```
t/rose-app-run/11-file-incr-2.t
```

### CODE REVIEW

File: lib/python/rose/config_processors/fileinstall.py [hunk 1]
```diff
@@ -253,22 +253,24 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
...
-                        ):
-                            target.is_out_of_date = True
-                    finally:
-                        conn.close()
+                    prev_dep_locs = conn.execute(
+                        """
+                            SELECT *
+                            FROM dep_names
+                            WHERE name=?
+                            ORDER BY ROWID
```
Comment (oliver-sanders) [inline lib/python/rose/config_processors/fileinstall.py:261]: This is needed to get the rows to come out in the right order.

File: lib/python/rose/config_processors/fileinstall.py [hunk 2]
```diff
@@ -253,22 +253,24 @@ def _process(self, conf_tree, nodes, loc_dao, **kwargs):
...
-                        prev_dep_locs = [
-                            loc_dao.select(i) for i in prev_dep_locs
-                        ]
-                        if (
-                            [i.name for i in prev_dep_locs] !=
-                            [i.name for i in target.dep_locs]
-                        ):
-                            target.is_out_of_date = True
-                    finally:
-                        conn.close()
```
Comment (oliver-sanders) [inline lib/python/rose/config_processors/fileinstall.py:271]: This is my bad, it turns out that this method can be called multiple times on the same DB (which I should have expected) so closing it here will cause subsequent errors. TBH I'm not sure if/how the DB is closed.

**Comment by oliver-sanders** [comment]
The `t/rose-app-run/` tests hit this logic pretty hard and are a good benchmark of functionality.

**Review by wxtim — APPROVED** [review]
- [x] Read and sanity checked.
- [x] Run tests locally.
- [x] Checked that test I recently wrote is still passing.

**Pull request merged**